rust/src/treefile.rs

```diff
@@ -1556,9 +1626,30 @@ etc-group-members:
         {
             let workdir = tempfile::tempdir().unwrap();
             let tf = new_test_treefile(workdir.path(), VALID_PRELUDE, None).unwrap();
-            let default_cfg = tf.get_check_groups();
-            assert_eq!(default_cfg.variant, CheckPasswdType::Previous);
-            assert_eq!(default_cfg.filename, None);
+            let default_cfg = tf.parsed.get_check_groups();
+            assert_eq!(default_cfg, &CheckGroups::Previous);
+        }
+        {
+            let input = VALID_PRELUDE.to_string() + r#"check-groups: { "type": "none" }"#;
+            let workdir = tempfile::tempdir().unwrap();
+            let tf = new_test_treefile(workdir.path(), &input, None).unwrap();
+            let custom_cfg = tf.parsed.get_check_groups();
+            assert_eq!(custom_cfg, &CheckGroups::None);
+        }
+        {
+            let input = VALID_PRELUDE.to_string()
+                + r#"check-groups: { "type": "data", "entries": { "bin": 1 } }"#;
+            let workdir = tempfile::tempdir().unwrap();
```

Comment:
I realize this isn't new but I prefer to have test functions return `Result<>` and hence this would be `tempfile::tempdir()?` the same as real code.  Go completely fails at this for example, forcing test functions to use special helpers around the `*testing.T` API.

Is there some tool that requires e.g. `// unwrap safety:` comments?  Would be nice.

Reply:
We went that path (Result and Try) in some Cincinnati-related codebases and my recommendation from such experience is to avoid doing that.
The rationale is that those test-unwraps should never fail, and when an unwrap unexpectedly fails in a remote CI you want to quickly know **both the error and the exact file/line** where it failed.
The naïf error handling approach via `?` discards fileline details, resulting in useless friction when inspecting remote CI failures.